src/core/apply.js

```diff
@@ -52,12 +54,24 @@ const callback = (obj, prop) => {
  * @since 0.4.0
  */
 const apply = (obj, path, operation) => {
-  const walkPath = (curObj, curPath) => {
+  const walkPath = (curObj, curPath, doCopy = true) => {
     const [prop, ...pathRest] = curPath
 
-    const value = callback(curObj, prop)
+    if (isSlice(prop)) {
+      const [start, end] = getSliceBounds(prop, length(curObj))
+
+      const newArr = copy(curObj, true)
```

Comment:
the second param is not clear. Usually the second param in a copy function is for deep not asArray

Reply:
As long as this is documented...

Reply:
And this is private code